packages/analytics-js-integrations/src/integrations/Mixpanel/util.js

```diff
@@ -241,21 +240,32 @@ const getConsolidatedPageCalls = config =>
     : true;
 
 /**
- * Generates a custom event name for a page calls.
+ * Generates a custom event name for a page or screen.
  *
  * @param {Object} message - The message object
  * @param {string} userDefinedEventTemplate - The user-defined event template to be used for generating the event name.
+ * @throws {ConfigurationError} If the event template is missing.
  * @returns {string} The generated custom event name.
+ * @example
+ * const userDefinedEventTemplate = "Viewed {{ category }} {{ name }} Page";
+ * const message = {name: 'Home', properties: {category: 'Index'}};
+ * output: "Viewed Index Home Page"
  */
 const generatePageCustomEventName = (message, userDefinedEventTemplate) => {
-  try {
-    const eventTemplate = Handlebars.compile(userDefinedEventTemplate);
-    return eventTemplate(message);
-  } catch (error) {
-    logger.error(`Error generating custom event name: ${error.message}`);
-    // Return a default or empty string as a fallback
-    return 'Loaded a Page';
+  let eventName = userDefinedEventTemplate
+    .replace('{{ category }}', message.properties?.category || '')
+    .trim();
+  eventName = eventName.replace('{{ name }}', message.name || '').trim();
+  // Remove any extra space between placeholders
+  eventName = eventName.replace(/\s{2,}/g, ' ');
+
+  // Check if any placeholders remain
+  if (eventName.includes('{{')) {
+    // Handle the case where either name or category is missing
+    eventName = eventName.replace(/{{\s*\w+\s*}}/g, '');
   }
```

Comment:
If we take responsibility for doing this, we should also handle other scenarios like malformed templates (orphan `{{`), etc., so let's skip it.

Reply:
The onus is completely on the user to configure the template correctly.